_inc/lib/class.jetpack-iframe-embed.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Tweak the preview when rendered in an iframe
+ */
+
+class Jetpack_Iframe_Embed {
+	static function init() {
+		if ( ! self::is_embedding_in_iframe() ) {
+			return;
+		}
+
+		// Disable the admin bar
+		if ( ! defined( 'IFRAME_REQUEST' ) ) {
+			define( 'IFRAME_REQUEST', true );
+		}
+
+		// Prevent canonical redirects
+		remove_filter( 'template_redirect', 'redirect_canonical' );
+
+		add_action( 'wp_head', array( 'Jetpack_Iframe_Embed', 'noindex' ), 1 );
+		add_action( 'wp_head', array( 'Jetpack_Iframe_Embed', 'base_target_blank' ), 1 );
+
+		add_filter( 'shortcode_atts_video', array( 'Jetpack_Iframe_Embed', 'disable_autoplay' ) );
+		add_filter( 'shortcode_atts_audio', array( 'Jetpack_Iframe_Embed', 'disable_autoplay' ) );
+
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			wp_enqueue_script( 'jetpack-iframe-embed', WPMU_PLUGIN_URL . '/jetpack-iframe-embed/jetpack-iframe-embed.js', array( 'jquery' ) );
+		} else {
+			$ver = sprintf( '%s-%s', gmdate( 'oW' ), defined( 'JETPACK__VERSION' ) ? JETPACK__VERSION : '' );
+			wp_enqueue_script( 'jetpack-iframe-embed', '//s0.wp.com/wp-content/mu-plugins/jetpack-iframe-embed/jetpack-iframe-embed.js', array( 'jquery' ), $ver );
```

Comment:
Hi, this needs to be loaded in `<head>`, otherwise, the user would be able to quickly click links before this script loads and that will cause the preview to fail (show a blank page or something similar). This script will not be loaded for any visitor or even admin, visiting their site directly. The only case when we load this JS is when the site is embedded in Calypso.